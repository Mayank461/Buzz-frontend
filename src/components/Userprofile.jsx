import axios from 'axios';
import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { API_URL } from '../config';
import UserlistWidget from './UserlistWidget';

export default function Userprofile({ suggestFriend, myData, refresh }) {
  const [user, setUser] = useState({});
  const { id } = useParams();
  const [friendStatus, setFriendStatus] = useState('ADD FRIEND');

  let isFriend = myData.friends.myFriends
    .map(({ _id }) => _id === id)
    .includes(true);

  let isPending =
    myData.friends.mySentRequests.map(({ _id }) => _id === id).includes(true) ||
    myData.friends.myFriendRequests.map(({ _id }) => _id === id).includes(true);

  useEffect(() => {
    axios
      .get(`${API_URL}/users/` + id, { withCredentials: true })
      .then((res) => {
        setUser(res.data);
      })
      .catch((err) => console.log(err.message));
  }, []);

  function SendReq() {
    axios
      .get(`${API_URL}/users/sendRequest/` + id, {
        withCredentials: true,
      })
      .then((res) => {
        refresh();
        // alert(res.data.message);
      })
      .catch((err) => console.log(err.message));
  }

  function DeleteFriend() {
    axios
      .get(`${API_URL}/users/deleteRequest/` + id, {
        withCredentials: true,
      })
      .then((res) => {
        refresh();
        // alert(res.data.message);
      })
      .catch((err) => console.log(err.message));
  }

  return (
    <div className="container mt-3">
      <div className="row">
        <div className="col-md-9 bg-white">
          <div className="position-relative">
            <img
              src="https://images.unsplash.com/photo-1495277493816-4c359911b7f1?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1165&q=80"
              className="coverpic"
            />
            <img
              src={user.picture_url || require('../images/blank-profile.png')}
              className="size"
            />
          </div>
          <div className="mx-4 ">
            <h1 className="mt-5"> {"firstname" in user?user.firstname + ' ' + user.lastname:"Unknown User"}</h1>
            <p>{user.bio} </p>
            <p>
              {`${user?.city} , ${user?.state} , India | ${user?.friends?.myFriends?.length}  Friend`}
            </p>
            <div className="d-flex">
              {/* {console.log(myData?.friends?.myFriends[0], id)} */}
              {/* {console.log(myData.friends.myFriends[0])} */}

              {isFriend ? (
                <div onClick={DeleteFriend} className="btn btn-danger me-3">
                  REMOVE FRIEND
                </div>
              ) : isPending ? (
                <div className="btn btn-dark me-3">Request Pending</div>
              ) : (
                <div onClick={SendReq} className="btn btn-primary me-3">
                  {friendStatus}
                </div>
              )}
              <button className="border boder-white">Visit Website</button>
            </div>
          </div>
        </div>
        <div className="col-md-3 profile-sidebar">
          <UserlistWidget
            title="Friend Suggestions"
            friendList={suggestFriend}
          />
        </div>
      </div>
    </div>
  );
}
