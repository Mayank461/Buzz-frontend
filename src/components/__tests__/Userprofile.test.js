import {render,screen,cleanup} from '@testing-library/react';
import { useState } from 'react';
import Userprofile from "../Userprofile";

 

test('should render Userprofile Panel', () => {
    const user={firstname:"kumar",lastname:"prashant",friends:{myFriends:[],mySentRequests:[{}]}};
    const SFriend =[];
    const toggleRefresh = () => setRefresh((p) => !p);
   const {debug}= render( <Userprofile
        myData={user}
        suggestFriend={SFriend}
        refresh={toggleRefresh}
      />)
      const text = screen.queryByTestId('userfullname').innerHTML;
      expect(text).toBe("Unknown User");
})