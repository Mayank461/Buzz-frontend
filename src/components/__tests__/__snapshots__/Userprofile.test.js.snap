// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Userprofile Panel 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-8 col-lg-9 bg-white mt-md-3 pb-4 p-0 shadow-lg  bg-body rounded "
    >
      <div
        className="position-relative"
      >
        <img
          alt=""
          className="coverpic"
          src="https://images.unsplash.com/photo-1495277493816-4c359911b7f1?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1165&q=80"
        />
        <img
          alt=""
          className="size"
          src="blank-profile.png"
        />
      </div>
      <div
        className="mx-4 "
      >
        <h1
          className="mt-5"
          data-testid="userfullname"
        >
          Unknown User
        </h1>
        <p>
           
        </p>
        <p>
          undefined , undefined , India | undefined  Friend
        </p>
        <div
          className="d-flex"
        >
          <div
            className="btn btn-dark me-3"
          >
            Request Pending
          </div>
          <a
            className="btn btn-outline-dark"
            rel="noopener noreferrer"
            target="_blank"
          >
            Visit Website
          </a>
        </div>
      </div>
    </div>
    <div
      className="col-md-4 col-lg-3 profile-sidebar mt-3"
    >
      <div
        className="border p-2 mb-3 scroll bg-white shadow-lg p-3 bg-body rounded border-0 "
      >
        <div
          className="d-flex justify-content-between"
        >
          <h6>
            Friend Suggestions
          </h6>
        </div>
        <div
          className="text-center"
        />
      </div>
    </div>
  </div>
</div>
`;
